doc/buffers.rst

```diff
@@ -946,6 +946,203 @@ there must be no side effects between the check and the operation:
+
+* ``%TypedArray%.prototype.filter()`` is missing.
+
+* ``%TypedArray%.prototype.find()`` is missing.
+
+* ``%TypedArray%.prototype.findIndex()`` is missing.
+
+* ``%TypedArray%.prototype.forEach()`` is missing.
+
+* ``%TypedArray%.prototype.indexOf()`` is missing.
+
+* ``%TypedArray%.prototype.join()`` is missing.
+
+* ``%TypedArray%.prototype.keys()`` is missing.
+
+* ``%TypedArray%.prototype.lastIndexOf()`` is missing.
+
+* ``%TypedArray%.prototype.map()`` is missing.
+
+* ``%TypedArray%.prototype.reduce()`` is missing.
+
+* ``%TypedArray%.prototype.reduceRight()`` is missing.
+
+* ``%TypedArray%.prototype.reverse()`` is missing.
+
+* ``%TypedArray%.prototype.set()`` exists, semantics need to be checked.
+
+* ``%TypedArray%.prototype.slice()`` is missing.
+
+* ``%TypedArray%.prototype.some()`` is missing.
+
+* ``%TypedArray%.prototype.sort()`` is missing.
+
+* ``%TypedArray%.prototype.subarray()`` exists, semantics need to be checked.
+
+* ``%TypedArray%.prototype.toLocaleString()`` is missing.
+
+* ``%TypedArray%.prototype.toString()`` is missing.
+
+* ``%TypedArray%.prototype.values()`` is missing.
+
+* ``%TypedArray%.prototype[@@iterator]()`` is missing.
+
+* ``get %TypedArray%.prototype[@@toStringTag]`` is missing.
+
+The initial implementations for some of the missing methods can be the
+equivalent methods in ``Array.prototype`` with the caveat that ``.length``
+should be accessed directly without invoking side effects.  For now this
+would not be an issue because typed array ``.length`` is a virtual own
+property, and accessing it has no side effects.
+
+Update to newer Node.js Buffer API version
+------------------------------------------
+
+Current:
+
+  - https://nodejs.org/docs/v0.12.1/api/buffer.html
+
+Latest at time of writing:
+
+  - https://nodejs.org/docs/v6.7.0/api/buffer.html
+
+Gap between current implementation and latest:
+
+* ``Buffer.prototype`` should inherit from ``Uint8Array.prototype`` now;
+  Buffer instances are now Uint8Array instances, and
+  ``new Uint8Array([1,2,3]).__proto__ == Uint8Array.prototype`` is true.
+
+* Buffers can be for-of iterated; buf.values(), buf.keys(), and
+  buf.entries() create iterators.  For-of iteration requires @@iterator
+  support.
+
+* ``new Buffer(...)`` is deprecated in favor of ``Buffer.from()``.
+
+* ``new Buffer(arrayBuffer[, byteOffset [, length]])`` variant is not
+  supported.  This variant is already deprecated.
+
+* ``new Buffer(string[, encoding])`` does not handle encoding correctly.
+  The internal string representation (CESU-8, extended UTF-8, or even
+  invalid UTF-8) is used as buffer bytes as-is.  This is also incorrect
+  for Node.js Buffer v0.12.1 (and already incorrect in master).
+
+* ``Buffer.alloc(size[, fill[, encoding]])`` is missing.
+
+* ``Buffer.allocUnsafe(size)`` is missing.  In practice it should be
+  implemented as ``Buffer.alloc(size)`` (ignoring any further parameters)
+  so that a zero-filled buffer is allocated.
```

Comment:
Btw, I guess to be really pedantic, ES6 doesn't say that there isn't some out-of-spec way of creating ArrayBuffers which are not zeroed. IIRC it just says that *if* you create one using `new ArrayBuffer()` it's zero initialized.

And in fact Duktape allows this too: you can create an external buffer and create an ArrayBuffer over it, in effect creating an ArrayBuffer over whatever data you want (initialized or not).

Finally, the Node.js API just says Buffers are Uint8Arrays, but doesn't say where the underlying ArrayBuffer *must* come from.... :)